BaseApi.js

```diff
@@ -31,21 +31,17 @@ const getHeaders = (token) => {
   return headers;
 };
 
-const makePostRequest = (url, data = {}, token=null) => {
-  return fetch(url, {
+const makePostRequest = async (url, data = {}, token=null) => {
+  const response = await fetch(url, {
     method: 'POST',
     headers: getHeaders(token),
     body: JSON.stringify(data)
-  })
-    .then(response => {
-      const statusCode = response.status;
-      const data = response.json();
-      return Promise.all([statusCode, data]);
-    })
-    .then(values => {
-      return {'statusCode': values[0], 'data': values[1]};
-    })
-    .catch(error => console.error(error));
+  });
+
+  const statusCode = response.status;
+  const payload = response.json();
+  const values = await Promise.all([statusCode, payload]);
+  return {'statusCode': values[0], 'data': values[1]};
```

Comment:
Is this some different functionality than before or is it just syntax? As far as I understood, the 'await' and 'then' are kind of doing the same, is that right?

Reply:
Yes it is basically the same. Just syntactical sugar 😶‍🌫️
i thought i write `async/await` because the whole data passing chain i wrote it that way. Personally i think it is more readable than promise notation

Reply:
Ok, I dont mind, we can also use that notation, if you prefer it...But to be consistent, we should then also change it in all other places, so also for makeGetRequest and makePutRequest and all the methods that use it. 
And... is there a way to catch an error and log it?

Reply:
hm, okay maybe then i will  change that one to promise notation. think my slightly tendency to async await does not justify the change on a lot of spots in our code..hehe. especially we would need to re-test and verify all spots are still working after change.

with async/await we would need to use `try catch`. which also makes it a bit less readable at the end.